startup/GafferScene/pathMatcherCompatibility.py

```diff
@@ -0,0 +1,42 @@
+##########################################################################
+#
+#  Copyright (c) 2018, Image Engine Design Inc. All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of John Haddon nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import GafferScene
+import Gaffer
+
+GafferScene.PathMatcher = IECore.PathMatcher
```

Comment:
We're missing the `import IECore` for this. I think we're probably getting lucky because the `IECore.ConfigLoader` that loads the startup files [reuses the same contextDict](https://github.com/ImageEngine/cortex/blob/master/python/IECore/ConfigLoader.py#L72) for each startup file. I'd like to fix that, but I wouldn't like to bet it wouldn't break something in the startup sequence at IE - what's your feeling @andrewkaufman? If I made a PR would we give it a go?

Reply:
I have a feeling that is considered a feature by at least one essential and highly configured production software, @ivanimanishi to confirm.

Reply:
Yeah, I thought that might be the case. It does mean that errors like the one here can go silently unnoticed though, and then a change somewhere else breaks it.

Perhaps I could make a PR allow `contextDict = None` to be passed, signifying that there is no context to share, and a fresh one should be made per file. Then we could use it in Cortex/Gaffer startup without affecting any other projects...

Reply:
That sounds fine